src/runtime_src/core/include/xrt/experimental/xrt_module.h

```diff
@@ -107,13 +107,13 @@ class module : public detail::pimpl<module_impl>
    *
    * Copy content of existing module into an allocation associated
    * with the specified hardware context.
-   * If module has multiple control codes, index is used to identify
-   * the control code that needs to be run.
+   * If module has multiple control codes, ctrl code id is used to
+   * identify the control code that needs to be run.
    *
    * Throws if module is not compatible with hardware context
    */
   XRT_API_EXPORT
-  module(const xrt::module& parent, const xrt::hw_context& hwctx, uint32_t ctrl_code_idx);
+  module(const xrt::module& parent, const xrt::hw_context& hwctx, std::string ctrl_code_id);
```

Comment:
We are sure no binary in-the-field is using the removed ctor, right?

Reply:
This constructor is being called from copy_module function from xrt_kernel.cpp and no application code must be using it. So I guess this change should be safe Soren.

Reply:
I see, if it is internal, can't it be in module_int ?

Reply:
Yes Soren, removed the unused constructor from being public and gave undocumented version of construction using impl handle. Please review